cluster-service/cspr/resource-cleaner.yaml

```
@@ -0,0 +1,163 @@
+  labels:
+    azure.workload.identity/use: "true"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: resource-cleaner-cronjob
+subjects:
+- kind: ServiceAccount
+  name: resource-cleaner-cronjob
+  namespace: ${RESOURCE_CLEANER_NAMESPACE}
+roleRef:
+  kind: ClusterRole
+  name: ${RESOURCE_CLEANER_CLUSTERROLE_NAME}
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: resource-cleaner
+  namespace: ${RESOURCE_CLEANER_NAMESPACE}
+spec:
+  schedule: "* * * * *"  # Run every minute for debugging
+  successfulJobsHistoryLimit: 3
+  failedJobsHistoryLimit: 3
+  concurrencyPolicy: Forbid  # Don't run concurrent jobs
+  jobTemplate:
+    spec:
+      backoffLimit: 2  # Retry up to 2 times on failure
+      template:
+        metadata:
+          labels:
+            azure.workload.identity/use: "true"
+        spec:
+          serviceAccountName: resource-cleaner-cronjob
+          restartPolicy: Never
+          containers:
+          - name: resource-cleaner
+            image: ${AZURE_CLI_IMAGE}
+            imagePullPolicy: Always
+            command: ["/bin/bash"]
+            args:
+            - -c
+            - |
+              set -euo pipefail
+              
+              echo "========================================"
+              echo "ARO-HCP Resource Cleaner"
+              echo "Started at: $(date -u)"
+              echo "========================================"
+              
+              echo "Checking for required tools..."
+              
+              # Check if jq is available, if not download it
+              if ! command -v jq > /dev/null 2>&1; then
+                echo "jq not found, downloading static binary..."
+                curl -sL https://github.com/jqlang/jq/releases/download/jq-1.7.1/jq-linux-amd64 -o /usr/local/bin/jq
+                chmod +x /usr/local/bin/jq
+                echo "✓ jq installed"
+              else
+                echo "✓ jq already available"
+              fi
+              
+              # Verify all required tools
+              echo "Verifying required tools..."
+              for tool in az jq python3 curl bash; do
+                if ! command -v $tool > /dev/null 2>&1; then
+                  echo "ERROR: Required tool '$tool' is not available"
+                  exit 1
+                fi
+              done
+              echo "✓ All required tools verified"
+              
+              echo "Copying scripts to /tmp and making executable..."
+              cp -r /scripts/* /tmp/
+              cd /tmp
+              chmod +x *.sh
+              
+              echo "Logging into Azure using workload identity..."
+              az login --identity || {
+                echo "ERROR: Failed to authenticate with Azure"
+                echo "Ensure workload identity is properly configured"
+                exit 1
+              }
```

```suggestion
               az login --service-principal \
                --username "${AZURE_CLIENT_ID}" \
                --tenant "${AZURE_TENANT_ID}" \
                --federated-token "$(cat ${AZURE_FEDERATED_TOKEN_FILE})" || {
                echo "ERROR: Failed to authenticate with Azure"
                echo "Ensure workload identity is properly configured"
                exit 1
              }
```

We'll need to change this with this suggestion